crates/cxx-qt-gen/src/parser/property.rs

```diff
@@ -20,10 +87,84 @@ impl ParsedQProperty {
             let _comma = input.parse::<Token![,]>()?;
             let ident = input.parse()?;
 
-            // TODO: later we'll need to parse setters and getters here
-            // which are key-value, hence this not being parsed as a list
+            if input.is_empty() {
+                // No flags passed so desugar: ```#[qproperty(T, ident)]``` -> ```#[qproperty(T, ident, read, write, notify)]```
+                Ok(Self {
+                    ident,
+                    ty,
+                    flags: QPropertyFlags::default(),
+                })
+            } else {
+                let _comma = input.parse::<Token![,]>()?; // Start of final identifiers
+
+                let punctuated_flags: Punctuated<Meta, Token![,]> =
+                    Punctuated::parse_terminated(input)?;
+
+                // Remove the commas and collect the individual meta items
+                let flags: Vec<Meta> = punctuated_flags.clone().into_iter().collect();
+
+                let mut read = None;
+                let mut write = None;
+                let mut notify = None;
 
-            Ok(Self { ident, ty })
+                // Create mutable closure to capture the variables for setting with the Meta values
+                let mut update_fields = |ident: &Ident, value: Option<Ident>| -> Result<()> {
+                    let variable = match ident.to_string().as_str() {
+                        "read" => &mut read,
+                        "write" => &mut write,
+                        "notify" => &mut notify,
+                        _ => {
+                            return Err(Error::new(
+                                ident.span(),
+                                "Invalid flag passed, must be one of read, write, notify",
+                            ));
+                        }
+                    };
+                    *variable = Some(value.map_or(FlagState::Auto, FlagState::Custom));
+
+                    Ok(())
+                };
+
+                for flag in flags {
+                    // Could maybe refactor parse_meta_name_value to parse_meta,
+                    // which would return an (Ident, Option<Ident>) and extract the logic below to a new fn
+                    match flag {
+                        Meta::Path(path) => {
+                            // Set flag as auto
+                            update_fields(path.require_ident()?, None)?;
+                        },
+                        Meta::NameValue(name_value) => {
+                            let kv_pair = parse_meta_name_value(&name_value)?;
```

Comment:
could unpack here with like `let (field, value) = parse_meta_name_value(...)` then you don't need to use `.1` and `.0` etc

Reply:
done